solr/solrj-streaming/src/java/org/apache/solr/client/solrj/io/stream/Facet2DStream.java

```diff
@@ -325,13 +322,7 @@ public void open() throws IOException {
     if (cache != null) {
       cloudSolrClient = cache.getCloudSolrClient(zkHost);
     } else {
-      final List<String> hosts = new ArrayList<>();
-      hosts.add(zkHost);
-      cloudSolrClient =
-          new CloudLegacySolrClient.Builder(hosts, Optional.empty())
-              .withSocketTimeout(30000, TimeUnit.MILLISECONDS)
-              .withConnectionTimeout(15000, TimeUnit.MILLISECONDS)
-              .build();
+      cloudSolrClient = SolrClientCache.newCloudHttp2SolrClient(zkHost, null);
```

Comment:
This line is curious... it seems you have made a public static method on SolrClientCache for creating a CloudHttp2SolrClient, thus making this class a builder for anyone anywhere who wants such a client?  Shouldn't we let the builder of CloudSolrClient do such a job?

Reply:
yes I agree with your concern. but I added it to replace all manual builder use from across the streaming module. there is no way to define visibility to module only, so this is what I ended up with. I prefer this single point of entry because it handles the timeout setting correctly, rather than repeating this code all over. the inconsistency was that some of the builders have increased timeouts, some don't, it's hard to understand what goes where in the streaming module.

Reply:
thinking about it a bit more, we can transition all streaming code to use 'client cache' instances instead of 'http clients' directly and this would allow all the builder code to be in one place with no public static methods.

Reply:
Yeah I think that's best.
And if there is timeout initialization handling in SolrClientCache, I wonder why it's there; why isn't it in the builder of CloudSolrClient etc.?  SCC should just be a cache IMO.  Nothing much to it.  CC @joel-bernstein

Reply:
the way I read it, this class attempts to set a min timeout(~60s) irrespective of what was set on client originally ... and also in passing manages to reset the timeouts back to 60s if they are bigger (which is an unfortunate side effect I wanted to address with this change) - I'm thinking the default httpclient on the UpdateShardHandler might play a central role here.

Reply:
@dsmiley did a complete pass over all classes and moved everything to cache. please take a look.